ext/zlib/zlib.c

```diff
@@ -917,8 +917,8 @@ PHP_FUNCTION(inflate_add)
 	}
 
 	if (!(ctx = zend_fetch_resource_ex(res, NULL, le_inflate))) {
-		php_error_docref(NULL, E_WARNING, "Invalid zlib.inflate resource");
-		RETURN_FALSE;
+		zend_type_error("Invalid zlib.inflate resource");
```

Comment:
Better to pass name to `zend_fetch_resource_ex` and let it throw the standard error. I also think that the `_ex` here is unnecessary.